client/my-sites/themes/theme-options.js

```diff
@@ -92,56 +96,106 @@ export const info = {
 		comment: 'label for displaying the theme info sheet'
 	} ),
 	icon: 'info',
-	getUrl: ( theme, site ) => getDetailsUrl( theme, site ), // TODO: Make this a selector
+	getUrl: getDetailsUrl,
 };
 
 export const support = {
 	label: i18n.translate( 'Setup' ),
 	icon: 'help',
-	getUrl: ( theme, site ) => getSupportUrl( theme, site ),
+	getUrl: getSupportUrl,
 	// We don't know where support docs for a given theme on a self-hosted WP install are.
-	hideForSite: ( { isJetpack = false } = {} ) => isJetpack,
-	hideForTheme: theme => ! isPremium( theme )
+	hideForSite: ( state, site ) => isJetpackSite( state, site ),
+	hideForTheme: ( state, theme ) => ! isPremium( theme )
 };
 
 export const help = {
 	label: i18n.translate( 'Support' ),
-	getUrl: ( theme, site ) => getHelpUrl( theme, site ),
+	getUrl: getHelpUrl,
 	// We don't know where support docs for a given theme on a self-hosted WP install are.
-	hideForSite: ( { isJetpack = false } = {} ) => isJetpack,
+	hideForSite: ( state, site ) => isJetpackSite( state, site ),
 };
 
-export function bindOptionToDispatch( option, source ) {
-	return dispatch => Object.assign(
-		{},
-		option,
-		option.action
-			? { action: bindActionCreators(
-				( theme, site ) => option.action( theme, site, source ),
-				dispatch
-				) }
-			: {}
-	);
-}
-
-export function bindOptionsToDispatch( options, source ) {
-	return dispatch => mapValues( options, option => bindOptionToDispatch( option, source )( dispatch ) );
-}
-
-// Ideally: same sig as mergeProps. stateProps, dispatchProps, ownProps
-function bindOptionToSite( option, site ) {
-	return Object.assign(
-		{},
-		option,
-		option.action
-			? { action: theme => option.action( theme, site ) }
-			: {},
-		option.getUrl
-			? { getUrl: theme => option.getUrl( theme, site ) }
-			: {}
-	);
-}
-
-export function bindOptionsToSite( options, site ) {
-	return mapValues( options, option => bindOptionToSite( option, site ) );
-}
+const ALL_THEME_OPTIONS = {
+	customize,
+	preview,
+	purchase,
+	activate,
+	tryandcustomize,
+	signup,
+	separator,
+	info,
+	support,
+	help
+};
+
+const ALL_THEME_ACTIONS = { activate: activateAction }; // All theme related actions available.
+
+export const connectOptions = connect(
+	( state, { options: optionNames, site } ) => {
+		let options = pick( ALL_THEME_OPTIONS, optionNames );
```

Comment:
The idea is that individual theme option objects -- used separately from `connectOptions` -- almost don't make any sense at all, since they'd always need to be bound at least to `state` (and `dispatch`, in case of e.g. `activate` -- and `install` and `delete`, later on), and sometimes to `siteId`. So I'm thinking of option objects more of an implementation detail right now (the interface being `connectOptions`). I should probably stop `exporting` them.